07-Extract-NDVI-From-Rasters-in-R.Rmd

```diff
@@ -161,104 +166,122 @@ names(avg_NDVI_HARV) <- "meanNDVI"
+# allows us to search for more than one pattern in our text strings.
 julianDays <- gsub(pattern = "X|_HARV_ndvi_crop", #the pattern to find 
             x = row.names(avg_NDVI_HARV), #the object containing the strings
             replacement = "") #what to replace each instance of the pattern with
 
+#alternate format
+#julianDays <- gsub("X|_HARV_ndvi_crop", "", row.names(avg_NDVI_HARV))
+
 #make sure output looks ok
 head(julianDays)
 
-#add julianDay values as a column in the dataframe
+#add julianDay values as a column in the data frame
 avg_NDVI_HARV$julianDay <- julianDays
 
 #what class is the new column
 class(avg_NDVI_HARV$julianDay)
-
-#convert julian days into a time class
-#avg_NDVI_HARV$julianDay <- as.POSIXlt(avg_NDVI_HARV$julianDay, format="%j")
-
-#the code above does weird things. for one i think jday is 0 based indexing
-#but also it seems to want a year and defaults to 2015. i am not sure how to force 
-#it to 2011 which is when the data were collected
-#being lazy and useing int for the time being.
-avg_NDVI_HARV$julianDay <- as.integer(avg_NDVI_HARV$julianDay)
-
-# Parse that character vector
-#strptime(date_info, "%Y %j")
 ```
 
-#Scale Data
-The last thing that we need to do it so scale our data. Valid NDVI values range 
-between 0-1. The metadata included a `ScaleFactor`. NDVI data are scaled by a 
-factor of 10,000. A scale factor is commonly used in larger datasets. Storing decimal
-places actually consumes more space and creates larger file sizes compared to 
-storing integer values.
-
-```{r scale-data }
 
-#scale data by 10,000
-avg_NDVI_HARV$meanNDVI <- avg_NDVI_HARV$meanNDVI / 10000
-#view output
-head(avg_NDVI_HARV)
+##Create Date Column from Julian Day
+Currently the Julian day is recognized as a character class value.  We'd prefer
+if it was recognized by `R` as one of the data-time classes so that `R` can
+interpret and format it as a date when plotting and in other functions. 
 
-```
+For more information on date-time classes, see the NEON Data Skills lesson 
+[Convert Date & Time Data from Character Class to Date-Time Class (POSIX) in R ]({{ site.baseurl}} /R/Time-Series-Convert-Date-Time-Class-POSIX/).  
 
-#Plot NDVI Using GGPLOT
+We know our data is from 2011, therefore the first day or origin for our Julian
+day count is 01 January 2011.  After setting this as the origin we just need to 
+add the Julian day (as an integer) to the origin date.  Since the origin data
+was originally set as a Date class object, the new `Date` column is also Date 
+class. 
+``` {r convert-jd}
+#set the origin for the julian date (1 Jan 2011)
+origin<-as.Date ("2011-01-01")
 
-We now have a clean `data.frame` with properly scaled NDVI and julian days.
-Let's plot our data.
+#convert "julianDay" from class character to integer
+avg_NDVI_HARV$julianDay <- as.integer(avg_NDVI_HARV$julianDay)
 
-```{r plot-data }
+#create a date column, -1 added because origin is the 1st.  
```

Comment:
If origin date is 01/01/2011 and we say add Julian day 5 to it the answer is 01/06/2011, not 01/05/2011.  

Changed note:
#create a date column; -1 added because origin is the 1st. 
# If not -1, 01/01/2011 + 5 = 01/06/2011 which is Julian day 6, not 5.